table/server/underdb/glue/src/main/java/alluxio/table/under/glue/GlueDatabase.java

```diff
@@ -0,0 +1,367 @@
+    try {
+      GetTableRequest tableRequest = new GetTableRequest()
+          .withCatalogId(mGlueConfiguration.get(Property.CATALOG_ID))
+          .withDatabaseName(mGlueDbName)
+          .withName(tableName);
+      table = getClient().getTable(tableRequest).getTable();
+
+      List<Partition> partitions = batchGetPartitions(getClient(), tableName);
+      PathTranslator pathTranslator = mountAlluxioPaths(table, partitions);
+
+      // Glue does not provide column statistic information
+      List<ColumnStatisticsInfo> columnStatisticsData = new ArrayList<>();
+
+      PartitionInfo partitionInfo = PartitionInfo.newBuilder()
+          .setDbName(mGlueDbName)
+          .setTableName(tableName)
+          .addAllDataCols(GlueUtils.toProto(table.getStorageDescriptor().getColumns()))
+          .setStorage(GlueUtils.toProto(table.getStorageDescriptor(), pathTranslator))
+          .putAllParameters(table.getParameters())
+          .build();
+
+      Layout layout = Layout.newBuilder()
+          .setLayoutType(GlueLayoutFactory.TYPE)
+          .setLayoutData(partitionInfo.toByteString())
+          .build();
+
+      return new GlueTable(this,
+          pathTranslator,
+          tableName,
+          GlueUtils.toProtoSchema(table.getStorageDescriptor().getColumns()),
+          columnStatisticsData,
+          // Glue does not provide FieldSchema from API directly
+          // Get FieldSchema from storage description
+          GlueUtils.toProto(table.getPartitionKeys()),
+          partitions,
+          layout,
+          table);
+    } catch (EntityNotFoundException e) {
+      throw new NotFoundException("Table " + tableName + " does not exist.", e);
+    } catch (ValidationException e) {
+      e.printStackTrace();
+      throw new IOException("Failed to get table: "
+          + tableName + " with validation error: " + e.getMessage(), e);
+    } catch (GlueEncryptionException e) {
+      throw new IOException("Failed to get table: " + tableName + " error: " + e.getMessage(), e);
+    }
+  }
+
+  private List<Partition> batchGetPartitions(AWSGlueAsync glueClient, String tableName)
+      throws IOException {
+    List<Partition> partitions = new ArrayList<>();
+    try {
+      GetPartitionsRequest getPartitionsRequest =
+          new GetPartitionsRequest()
+              .withCatalogId(mGlueConfiguration.get(Property.CATALOG_ID))
+              .withDatabaseName(mGlueDbName)
+              .withTableName(tableName);
+      if (glueClient.getPartitions(getPartitionsRequest).getPartitions() != null) {
+        partitions = glueClient.getPartitions(getPartitionsRequest).getPartitions();
+      }
+      return partitions;
+    } catch (AWSGlueException e) {
+      throw new IOException("WARNING: Cannot get partition information for table: "
+          + tableName + ". error: " + e.getMessage(), e);
+    }
+  }
+
+  /**
+   * Get Glue Client.
+   *
+   * @return async glue client
+   */
+  public AWSGlueAsync getClient() {
```

Comment:
one thing i noticed when doing the Hive version of this, is that the client sometimes does not handle disconnection well. Have you tested the case where the connection is not always stable?